adapter/src/ethereum.rs

```diff
@@ -269,6 +278,71 @@ impl Adapter for EthereumAdapter {
         ewt_sign(&wallet, &self.keystore_pwd, &payload)
             .map_err(|err| AdapterError::Adapter(Error::SignMessage(err).into()))
     }
+
+    async fn get_deposit(
+        &self,
+        channel: &ChannelV5,
+        address: &Address,
+    ) -> AdapterResult<Deposit, Self::AdapterError> {
+        let outpace_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &OUTPACE_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let erc20_contract =
+            Contract::from_json(self.web3.eth(), channel.token.as_bytes().into(), &ERC20_ABI)
+                .map_err(Error::ContractInitialization)?;
+
+        let total: H256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(outpace_contract.query(
+                "deposits",
+                (
+                    H256(*channel.id()).into_token(),
+                    H160(*address.as_bytes()).into_token(),
+                ),
+                None,
+                Options::default(),
+                None,
+            ))
+            .await
+        })
+        .await
+        .map_err(Error::ContractQuerying)?;
+
+        let mut total = BigNum::from_str(&total.to_string())?;
+
+        let still_on_create_2: H256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(erc20_contract.query(
+                "balanceOf",
+                Bytes(channel.token.as_bytes().to_vec()).into_token(),
+                None,
+                Options::default(),
+                None,
+            ))
+            .await
+        })
+        .await
+        .map_err(Error::ContractQuerying)?;
+
+        let still_on_create_2 = BigNum::from_str(&still_on_create_2.to_string())?;
```

Comment:
you can use `still_on_create_2.to_string().parse()` this is what `FromStr` allows you to do. You can use it directly like this, but it's meant more like a syntax sugar for String & &str `.parse()`